org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/actions/ToggleBreakpointAdapter.java

```diff
@@ -1116,7 +1179,22 @@ static IStatus doToggleWatchpoints(IWorkbenchPart part, ISelection finalSelectio
 				BreakpointUtils.addJavaBreakpointAttributes(attributes, javaField);
 				resource = BreakpointUtils.getBreakpointResource(type);
 			}
-			JDIDebugModel.createWatchpoint(resource, typeName, fieldName, -1, start, end, 0, true, attributes);
+
+			IJavaWatchpoint watchPoint = JDIDebugModel.createWatchpoint(resource, typeName, fieldName, -1, start, end, 0, true, attributes);
+
+			if (BreakpointToggleUtils.isTriggerpoints()) {
+
+				watchPoint.setTriggerPoint(true);
+				BreakpointToggleUtils.setTriggerpoints(false);
+
+			}
+
+			if (BreakpointToggleUtils.isHitpoint()) {
+
+				watchPoint.setHitCount(10);
```

Comment:
When is this code block executed and why it has "10" hard coded?

Reply:
Removed the hard coded contents